src/client/eppo-client.ts

```diff
@@ -589,21 +590,43 @@ export default class EppoClient {
       // Check if the assigned variation is an active bandit
       // Note: the reason for non-bandit assignments include the subject being bucketed into a non-bandit variation or
       // a rollout having been done.
-      const banditVariations = this.banditVariationConfigurationStore?.get(flagKey);
-      const banditKey = banditVariations?.find(
-        (banditVariation) => banditVariation.variationValue === variation,
-      )?.key;
-
-      if (banditKey) {
-        evaluationDetails.banditKey = banditKey;
-        action = this.evaluateBanditAction(
+      const bandit = this.findBanditByVariation(flagKey, variation);
+      if (bandit) {
```

Comment:
this is a style suggestion, but would you consider an early exit so that the code below does not need to be indented.

```
if (!bandit) {
  // handle
}

evaluationDetails.banditKey = bandit.banditKey;
...
```

Reply:
I like it! Made the change